### ISSUE TITLE
Explain when to use `index` routes

### ISSUE THREAD

**Issue by jenweber** [ember-learn/guides-source#156]
**Problem:**
It's unclear what to actually use index routes for. We can add some more details to the Guides on this.

**To do:**
- add more detail to the [Index Routes](https://guides.emberjs.com/release/routing/defining-your-routes/#toc_index-routes) section. Only make changes to the markdown file for the latest version of Ember.

**Suggestions collected from Discord chat:**
- "I explain this by relating it to index.html. If you make a request to https://emberjs.com, what file are you requesting?  -- index. This has helped people understand why it's implicit (we don't need this.route('index', { path: ''}), and why it exists at every non-leaf-level node of the routing hierarchy."
- Be precise with talking about parent/child hierarchy. Technically an index route is a child route, like `posts/index` and `posts/1` so it's inaccurate to say that an index is a place to put content that should not be rendered on child routes. (Although that's my mental model)
- "what makes index special? It's implicitly created for you whenever there are child routes in your router.js. It has a path of '', and is thus great for rendering stuff on "folders" within your routing hierarchy"

maybe instead we could do something like:

Here's a routing structure with 
```
/posts/123/author
/posts/123/comments
```

What happens when a user visits /posts/123? We'll need to create a new route for this. Let's call it index for now.

```
this.route('post', { path: 'posts/:id' }, function () {
   this.route('index', { path: '' });
});
```

(some development of templates/post/index.hbs)
Hey, it turns out, we can remove this thing from router.js. Ember gives it to you for free
basically make it explicit, and then demonstrate that it's implicit
easier to explain while it's explicit

**How you know you have improved it**
Someone asked this question on Stack Overflow, and there was no great place to link them to in the Guides for an answer. If your work would help them, it's an improvement!

> I have a quick question Here, I am new to Ember and it looks really exciting, I would like to have multiple routes in my web application, but when I click on a route it redirects the outlet to ONLY this NEW page "route", I don’t want the old page + content of the new page.. Any Ideas how to do that?

**Comment by LalithaRajanala** [comment]
@jenweber : Are you still looking for volunteers for this? I would love to help.

**Comment by sivakumar-kailasam** [comment]
@LalithaRajanala we're always looking for volunteers 🙂. Feel free to claim & start working on this. You can always reach out to us on the #dev-learning channel on [discord](Ember Community Discord)

**Comment by jenweber** [comment]
Close by #222, thanks!
